### DISCUSSION TITLE
Variable length recording with a trigger that depends on a second trigger..

### DISCUSSION THREAD

**Discussion by inchinn1** [bonsai-rx/bonsai#1484] [Q&A] [answered]
I have task in which I am unsure of the best approach. I am trying to record a video 10s before a stimulus triggered via the digital pin and then for 10s after the stimulus, but with a condition that the trigger should wait more that 10s if a second signal is not yet emitted. (The key downs are serving as triggers for testing purposes) . Rather than skipUntil I think I need something like 'proceed if true'. Or should skip until work in this case?
![image](https://github.com/bonsai-rx/bonsai/assets/21177232/f471b3c1-793e-4788-849f-57c9b35babb6)
So if the second signal takes longer than 10s the length of the video would be longer

**Comment by bruno-f-cruz** [comment]
Hi,
I did not understand what you meant by "...but with a condition that the trigger should wait more that 10s if a second signal is not yet emitted.". Otherwise, check a possibly relevant solution attached

[1484.zip](https://github.com/bonsai-rx/bonsai/files/12110255/1484.zip)

**Reply by inchinn1** [reply]
Thanks, though I don't quite understand all of the sketch, I will have to read up on subjects. 
The overall goal is to record video either side of giving a stimulus, a minimum of 10s before and a fixed 10s after the stimulus. The stimulus is only given when an other signal is emitted (i.e. a treadmill not moving within the last 10s). So once video starts being recorded and the treadmill moves, prestimulus video just keeps getting recorded until the treadmill is still for at least 10s, then it proceeds with the stimulus and a further 10s of video capture. 
I have already figured out how to check for a non moving stimulus, I am just using keydowns for testing the recording part.
![image](https://github.com/bonsai-rx/bonsai/assets/21177232/93d0d951-25a0-4890-ae68-33e7f34c392f)

**Reply by inchinn1** [reply]
Is there something that will wait for 2 inputs to be true before proceeding?

**Reply by inchinn1** [reply]
It sounds like 'Gate' should do this but it doesn't seem to work...

**Reply by inchinn1** [reply, marked as answer]
Ah skip until does work I just needed to flip the inputs.
